src/main/kotlin/sidechain/iroha/IrohaChainHandler.kt

```diff
@@ -8,20 +9,26 @@ import sidechain.SideChainEvent
 /**
  * Dummy implementation of [ChainHandler] with effective dependencies
  */
-class IrohaChainHandler : ChainHandler<IrohaBlockStub> {
+class IrohaChainHandler : ChainHandler<iroha.protocol.BlockOuterClass.Block> {
 
     /**
      * TODO Replace dummy with effective implementation
      */
-    override fun parseBlock(block: IrohaBlockStub): List<SideChainEvent> {
+    override fun parseBlock(block: iroha.protocol.BlockOuterClass.Block): List<SideChainEvent> {
         logger.info { "Iroha chain handler" }
-        return block.transactions
-            .flatMap { it.commands }
-            .filter { it is IrohaCommand.CommandAddPeer }
+        return block.payload.transactionsList
+            .flatMap { it.payload.commandsList }
             .map {
-                it as IrohaCommand.CommandAddPeer
-                SideChainEvent.IrohaEvent.OnIrohaAddPeer(it.address, it.peerKey)
+                when {
+                    it.hasAddPeer() -> {
+                        val modelCmd = IrohaCommand.CommandAddPeer.fromProto(it.toByteArray())
```

Comment:
[See here](https://github.com/d3ledger/notary/pull/53/files#diff-9259c50e51838bd561e3f9da089cd562R51). It would be nice to call `fromProto()` on `SideChainEvent.IrohaEvent.OnIrohaAddPeer` and do not use `IrohaCommand.CommandAddPeer` here.

Reply:
Please, add tx hash in `SideChainEvent .IrohaEvent`.